File: src/main/java/sprykerFileCreator/model/definition/spryker/ZedFacadeDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedFacadeDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_FACADE;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}Facade";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Business";
    }

    @Override
    public String getMethodForReturnType() {
        return "getFacade()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_BUSINESS_FACTORY
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\Business\\AbstractFacade";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ZedFacadeDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ZedFacadeDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_FACADE
    override val namePattern: String
        get() = "{{ bundleName }}Facade"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Business"
    override val methodForReturnType: String
        get() = "getFacade()"
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.ZED_BUSINESS_FACTORY
        )
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\Business\\AbstractFacade"
}